DOMImplementation::createDocument should set content type based on namespace

cjkenn opened the pull request:
Set document content type based on the namespace.
Standard: https://dom.spec.whatwg.org/#dom-domimplementation-createdocument
Fixes #10743. 

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/10830)

<!-- Reviewable:end -->

cjkenn committed:
[10743] Add content type to XmlDocument constructor based on namespace

[10743] Fix namespace in createDocument test

[10743] Remove test ini file, match returns static strings instead of DOMString.

[10743] Fix arguments to XMLDocument::new

Update failing test

[10743] Add content type to XmlDocument constructor based on namespace

[10743] Fix namespace in createDocument test

[10743] Remove test ini file, match returns static strings instead of DOMString.

[10743] Fix arguments to XMLDocument::new

Update failing test

highfive commented:
Thanks for the pull request, and welcome! The Servo team is excited to review your changes, and you should hear from @metajack (or someone else) soon.

highfive commented:
Heads up! This PR modifies the following files:
 * @kichjang: components/script/dom/domimplementation.rs

Ms2ger commented:
Thanks for your PR.

You'll need to update `./tests/wpt/metadata/dom/nodes/DOMImplementation-createDocument.html.ini` (probably by removing it entirely).

-S-awaiting-review +S-needs-code-changes 

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/script/dom/domimplementation.rs, line 76 \[r2\]](https://reviewable.io:443/reviews/servo/servo/10830#-KGBdW2AIaeIDh5dzgms:-KGBdW2AIaeIDh5dzgmt:-68111357) ([raw file](https://github.com/servo/servo/blob/be091696c54c94c971a04e3e0ec260a6ece38c62/components/script/dom/domimplementation.rs#L76)):*
Maybe move the `DOMString::from` call into the `XMLDocument::new` call, so this match just has the static strings.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10830#-:-KGBdehXW0lWosQAawVd:530126996)*
<!-- Sent from Reviewable.io -->

cjkenn commented:
Ahhh I didn't realize there was a config file for expected test failures. Since all the tests pass, I will remove it.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10830#-:-KGCkw_GNrMUjcLN-9aG:-125209229)*
<!-- Sent from Reviewable.io -->

Ms2ger commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/script/dom/domimplementation.rs, line 84 \[r3\]](https://reviewable.io:443/reviews/servo/servo/10830#-KGGeJEdERT4zx7BtCqk:-KGGeJEeLXIxKX5RLidN:1521359015) ([raw file](https://github.com/servo/servo/blob/351a697d2c3124a685a4e7fa8cac4ba2daa19206/components/script/dom/domimplementation.rs#L84)):*
I'm sorry, I didn't explain that very well. I'd like you to write `Some(DOMString::from(content_type))` here, and not change the `XMLDocument::new` API itself.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10830)*
<!-- Sent from Reviewable.io -->

cjkenn commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/script/dom/domimplementation.rs, line 84 \[r3\]](https://reviewable.io:443/reviews/servo/servo/10830#-KGGeJEdERT4zx7BtCqk:-KGKKOevJ747G90WZBIz:-2016371997) ([raw file](https://github.com/servo/servo/blob/351a697d2c3124a685a4e7fa8cac4ba2daa19206/components/script/dom/domimplementation.rs#L84)):*
I really should have clarified that before I made the change! I did feel awkward doing it that way >_<

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10830)*
<!-- Sent from Reviewable.io -->

Ms2ger commented:
Thanks! This is good to go once you squash the commits into one; let me know if you need help with that.

-S-awaiting-review +S-needs-squash 

---

Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10830#-:-KGLUsAlwCgPxxETQuJr:167463354)*
<!-- Sent from Reviewable.io -->

cjkenn commented:
Squashed and ready to go!

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/10830#-:-KGQ4LQb3i2LkaKpTI0_:-479217129)*
<!-- Sent from Reviewable.io -->

KiChjang commented:
@bors-servo r=Ms2ger

bors-servo commented:
:pushpin: Commit fb1ac9c has been approved by `Ms2ger`

<!-- @bors-servo r=Ms2ger fb1ac9c530032b33a210ab6513ed2e145f2ae935 -->

bors-servo commented:
:hourglass: Testing commit fb1ac9c with merge e28cab5...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/634)

highfive commented:
▶ Unexpected subtest result in /dom/nodes/Document-contentType/contentType/createDocument.html:
      │ FAIL [expected PASS] document.implementation.createDocument: document.contentType === &#39;application/xml&#39;
      │   → assert_equals: expected &#34;application/xml&#34; but got &#34;application/xhtml+xml&#34;
      │ 
      │ @http://web-platform.test:8000/dom/nodes/Document-contentType/contentType/createDocument.html:3:3
      │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1397:20
      │ test@http://web-platform.test:8000/resources/testharness.js:495:9
      └ @http://web-platform.test:8000/dom/nodes/Document-contentType/contentType/createDocument.html:1:1

Ms2ger commented:
You'll have to update `tests/wpt/web-platform-tests/dom/nodes/Document-contentType/contentType/createDocument.html` too.

cjkenn commented:
Updated that test file and squashed again. I didn't run the full wpt-test suite, as I don't have enough time to wait for it (that or I cannot figure out how to run it faster). I checked a few tests in areas where they might fail because of this and they were passing, so hopefully there is no more failures...

KiChjang commented:
@bors-servo r=Ms2ger

bors-servo commented:
:pushpin: Commit d9128fb has been approved by `Ms2ger`

<!-- @bors-servo r=Ms2ger d9128fba07c65bde012c3fb9f5948386f4f3da39 -->

bors-servo commented:
:hourglass: Testing commit d9128fb with merge 408f9ab...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/600), [arm32](http://build.servo.org/builders/arm32/builds/636), [arm64](http://build.servo.org/builders/arm64/builds/621), [linux-dev](http://build.servo.org/builders/linux-dev/builds/632), [linux-rel](http://build.servo.org/builders/linux-rel/builds/680), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/648), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/629), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/658), [windows](http://build.servo.org/builders/windows/builds/318)

Ms2ger commented:
@cjkenn thanks! We don't expect you to run all of wpt locally; that's what our bits are for :)
